ggml/src/ggml-metal/ggml-metal.m

```diff
@@ -1126,7 +1126,16 @@ static bool ggml_metal_supports_op(const struct ggml_backend_metal_device_contex
         case GGML_OP_ARGMAX:
         case GGML_OP_NORM:
         case GGML_OP_ROPE:
-            return true;
+            {
+                const int mode = ((const int32_t *) op->op_params)[2];
+                if (mode && GGML_ROPE_TYPE_MROPE) {
+                    return false;
+                }
+                if (mode && GGML_ROPE_TYPE_VISION) {
+                    return false;
+                }
+                return true;
+            }
```

Comment:
You are missing a `return true;` before the `GGML_OP_ROPE` case. Without it, this code will by executed for the `GGML_OP_NORM`, `GGML_OR_ARGMAX`, etc. ops which is incorrect. Same for the `kompute` backend.

Reply:
Oh, I see. There are multiple `GGML_OP` cases sharing the same return statement.